freshmaker/pyxis.py

```diff
@@ -0,0 +1,205 @@
+        """
+        entity_url = '{}/{}'.format(self._api_root, entity)
+        default_params = {"page_size": "100"}
+        default_params.update(params)
+
+        response = requests.get(entity_url, params=default_params,
+                                auth=HTTPKerberosAuth(mutual_authentication=OPTIONAL))
+
+        if response.status_code == HTTPStatus.OK:
+            return response.json()
+
+        # Warn early, in case there is an error in the error handling code below
+        log.warning("Request to %s gave %r", response.request.url, response)
+
+        raise PyxisRequestError(response.status_code, response.json())
+
+    def _pagination(self, entity, params):
+        """
+        Process all pages in Pyxis
+
+        :param str entity: what data/entity to request from Pyxis
+        :param dict params: parameters to add to GET request
+        :return: list of all 'data' fields from responses from Pyxis
+        :rtype: list
+        """
+        ret = []
+        response_data = self._make_request(entity, params=params)
+        ret.extend(response_data.get('data', []))
+        page = 1
+        last_page = ceil(response_data['total'] / response_data['page_size'])
+        while page < last_page:
+            params.update({"page": page})
+            response_data = self._make_request(entity, params=params)
+            ret.extend(response_data.get('data', []))
+            page += 1
+        return ret
+
+    def get_operator_indices(self, params={}):
+        """
+        Get all index images for organization(s)(configurable) from Pyxis
+
+        :param dict params: parameters to add to GET request
+        """
+        organization = conf.pyxis_index_image_organization
+        params.update({"filter": "organization=="+organization})
+
+        entity = "operators/indices"
+        indices = self._pagination(entity, params)
+        return indices
+
+    def _get_bundles_per_index_image(self, index_images, params={}):
+        """
+        Get bundle images for all index images
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: bundle images per index image
+        :rtype: dict
+        """
+        bundles_per_index_image = {}
+        # we need 'bundle_path_digest' to find ContainerImage of that bundle
+        include_fields = self._default_include_fields + \
+                         ['data.channel_name', 'data.version',
+                          'data.related_images', 'data.bundle_path_digest']
+        for index_image in index_images:
+            path = index_image.get('path', '')
+            if not path:
+                continue
+            params.update({"filter": f"source_index_container_path=={path}",
+                           "include": ','.join(include_fields)})
+            bundles = self._pagination('operators/bundles', params)
+            bundles_per_index_image.update({path: bundles})
```

Comment:
There's no need to create a temporary dictionary just to assign a single value.
```suggestion
            bundles_per_index_image[path] = bundles
```